Looks great! Thanks for contributing.
Could you take a look at non-Windows builds? Looks like somethings wrong there:
```c++
../../../src/platform/windows/input/key.h:40:22: error: extra qualification on member 'getName'
    const char* Key::getName();
```

One thing I don't like is how Soft/Hard reset is handled, but to be honest, it's a limitation of the current key mapping system - `keyboard|F2` doesn't allow you to specify modifiers like Shift.

Also, it won't work if somebody was to map a hotkey to something else than a keyboard - for example, I could map `quick save/load` to a game controller, but that would fail due to hardcoded keyboard event check
```c++
if (event.key.keysym.sym == Key(config.hotkeys["quick_load"]).key)
```
There are some modifications needed to `SdlInputManager` to handle it, but for now, we can leave it as it.

--- a/src/platform/windows/gui/options/options.cpp
+++ b/src/platform/windows/gui/options/options.cpp
@@ -225,7 +225,7 @@ void button(const std::string& button, const char* tooltip, int controller = 0) 
             ImGui::OpenPopup("Waiting for key...");
         }
         ImGui::SameLine();
-        ImGui::Text(tooltip);
+        ImGui::TextUnformatted(tooltip);
 
         return;
     }

--- a/src/platform/windows/input/key.h
+++ b/src/platform/windows/input/key.h
@@ -37,7 +37,7 @@ struct Key {
     static Key mouseButton(SDL_MouseButtonEvent button);
     static Key controllerMove(SDL_ControllerAxisEvent axis);
     static Key controllerButton(SDL_ControllerButtonEvent button);
-    const char* Key::getName();
+    const char* getName();
 
     bool operator==(const Key& rhs) const;
 

